tests/aws-cpp-sdk-dynamodb-integration-tests/TableOperationTest.cpp

```diff
@@ -459,6 +448,427 @@ TEST_F(TableOperationTest, TestThrottling)
+            updateItemRequest.AddKey(HASH_KEY_NAME, AttributeValue(ss.str()));
+            ss.str("");
+            AttributeValueUpdate testValueAttribute;
+            ss << testValueColumnName << i * 2;
+            testValueAttribute.SetAction(AttributeAction::PUT);
+            AttributeValue valueAttribute;
+            valueAttribute.SetS(ss.str());
+            testValueAttribute.SetValue(valueAttribute);
+            updateItemRequest.AddAttributeUpdates(testValueColumnName, testValueAttribute);
+            ss.str("");
+            contextSp->updateItemOutcomes.push_back(m_client->UpdateItemCallable(updateItemRequest));
+        }
+
+        //wait for operations to finish.
+        for (auto& updateItemOutcome : contextSp->updateItemOutcomes)
+        {
+            updateItemOutcome.get();
+        }
+        return true;
+    };
+
+
+    retryObj.addFunction(RetryPlanner<TestContext>::FunctionBlock{
+        "updateItemRequests",
+        updateItemFunc,
+        0,
+        true
+    });
+
+    auto getItemFunc2 = [this, testValueColumnName, crudTestTableName, numItems , &requestIds](
+            std::shared_ptr<TestContext> contextSp, 
+            const Aws::String& id,
+            int numRetriesLeft)
+    {
+        AWS_UNREFERENCED_PARAM(id);
+        AWS_UNREFERENCED_PARAM(numRetriesLeft);
+
+        Aws::StringStream ss;
+        contextSp->getItemOutcomes.clear();
+        for (auto i : requestIds)
+        {
+            GetItemRequest getItemRequest;
+            ss << HASH_KEY_NAME << i;
+            AttributeValue hashKey;
+            hashKey.SetS(ss.str());
+            getItemRequest.AddKey(HASH_KEY_NAME, hashKey);
+            getItemRequest.SetTableName(crudTestTableName);
+
+            Aws::Vector<Aws::String> attributesToGet;
+            attributesToGet.push_back(HASH_KEY_NAME);
+            attributesToGet.push_back(testValueColumnName);
+            ss.str("");
+            contextSp->getItemOutcomes.push_back(m_client->GetItemCallable(getItemRequest));
+    
+        }
+
+        bool status = true;
+
+        for (auto it = requestIds.begin(); it != requestIds.end();)
+        {
+            auto i = *it;
+            GetItemOutcome outcome = contextSp->getItemOutcomes[i].get();
+            bool res = outcome.IsSuccess();
+            
+            AWS_EXPECT_SUCCESS(outcome);
+            GetItemResult result = outcome.GetResult();
+
+            Aws::StringStream hashKeyName;
+            hashKeyName << HASH_KEY_NAME << i;
+            Aws::Map<Aws::String, AttributeValue> returnedItemCollection = result.GetItem();
+
+            Aws::StringStream testValueColumn;
+            testValueColumn << testValueColumnName << i * 2;
+
+            res = res && (hashKeyName.str() == returnedItemCollection[HASH_KEY_NAME].GetS());
+            res = res && (testValueColumn.str() == returnedItemCollection[testValueColumnName].GetS());
+
+            if(numRetriesLeft > 0)
```

Comment:
Forced failure to try multiple times till exhaustion leading to success and subsequent GTEST MACRO evaluation